testgrid/cmd/entomologist/main.go

```diff
@@ -114,13 +128,13 @@ func main() {
 	defer cancelFunc()
 
 	client, _ := io.NewOpener(ctx, opt.gcsCredentials)
-	if client == nil {
-		logrus.Fatalf("Empty credentials (at %s) not allowed for write operation", opt.gcsCredentials)
-	}
 
 	// testgrid expects issue_state.proto files for an org/repo to be named this way
-	gcsFile := fmt.Sprintf("/bugs-%s-%s", opt.organization, opt.repository)
-	writer, _ := client.Writer(ctx, opt.gcsPath+gcsFile)
+	outputFile := fmt.Sprintf("%s/bugs-%s-%s", opt.output, opt.organization, opt.repository)
+	if fi, err := os.Stat(opt.output); err == nil && fi.Mode().IsRegular() {
```

Comment:
This might be a better place to put that logic.

You could have one stat call that checks the file:
- If the file exists, and it's a regular file, `outputFile=opt.output`
- If the file exists and is a directory, make a file there
- If it doesn't exist, try to create it.

I think the behavior you've implemented here is brilliant, but it took me a couple of readthroughs of the code to see that that's what was going on.

Reply:
I agree. The validation logic seems a little bit obscure and it would have been easier to understand with those three cases separately. Just one quick question.

I'm creating the file during option parsing to fail fast in case the file cannot be opened (e.g., /exists/noexists/file). I've seen that it can take a couple of minutes to parse a large project (kubernetes/kubernetes). Creating the file was just en "empirical" way of checking that the path provided in `--output` was valid.

Then, If I remove file creation during option parsing, I guess I cannot validate `--output`, right? (apart from the case where it is empty)? Like in [Configurator](https://github.com/kubernetes/test-infra/blob/2451842221e3a5a7923b544e9a739f8c7ddb5881/testgrid/cmd/configurator/main.go#L79). So I can:

1) Validate that `--output` is not empty. Add the logic that you commented here. CONS: writing can fail later.
2) Validate as it is done now but remove the file after it is created. Create the file again adding the logic that you commented. CONS: validate is much more obscure.

Thank you for your suggestions. I'm sorry if I'm getting a little complicated for a change that should be trivial.